pkgs/by-name/me/mealie/package.nix

```
@@ -0,0 +1,151 @@
+{ lib
+, callPackage
+, fetchFromGitHub
+, makeWrapper
+, nixosTests
+, python3
+, stdenv
+, writeShellScript
+}:
+
+let
+  version = "1.2.0";
+  src = fetchFromGitHub {
+    owner = "mealie-recipes";
+    repo = "mealie";
+    rev = "v${version}";
+    sha256 = "sha256-Kc49XDWcZLeJaYgiAO2/mHeVSOLMeiPr3U32e0IYfdU=";
+  };
+
+  frontend = callPackage (import ./mealie-frontend.nix src version) { };
+
+  python = python3.override {
+    packageOverrides = self: super: {
+      pydantic = self.pydantic_1;
+    };
+  };
+
+  crfpp = stdenv.mkDerivation {
+    pname = "mealie-crfpp";
+    version = "master";
+    src = fetchFromGitHub {
+      owner = "mealie-recipes";
+      repo = "crfpp";
+      rev = "c56dd9f29469c8a9f34456b8c0d6ae0476110516";
+      sha256 = "sha256-XNps3ZApU8m07bfPEnvip1w+3hLajdn9+L5+IpEaP0c=";
+    };
+  };
+
+in python.pkgs.buildPythonPackage rec {
+  pname = "mealie";
+  inherit version src;
+  pyproject = true;
+
+  patches = [
+    # See https://github.com/mealie-recipes/mealie/pull/3102
+    # Replace hardcoded paths in code with environment variables (meant for inside Docker only)
+    # So we can configure easily where the data is stored on the server
+    ./mealie_model_path.patch
+    ./mealie_init_db.patch
```

Now that you've upstreamed the patches: it's idiomatic to use `fetchpatch` to download the patches, rather than vendor them.

So something like:

```suggestion
    # https://github.com/mealie-recipes/mealie/pull/3102
    # Replace hardcoded paths in code with environment variables (meant for inside Docker only)
    # So we can configure easily where the data is stored on the server
    (fetchpatch {
      name = "mealie-model-path.patch";
      url = "https://github.com/mealie-recipes/mealie/commit/e445705c5d26b895d806b96b2f330d4e9aac3723.patch";
      hash = ""; # FIXME
    })
    (fetchpatch {
      name = "mealie-init-db.patch";
      url = "https://github.com/mealie-recipes/mealie/commit/06c528bfac0708af66aa0629f2e2232ddf07768f.patch";
      hash = ""; # FIXME
    })
```

And delete those two files.